### DISCUSSION TITLE
More events and states for fabric.Control

### DISCUSSION THREAD

**Discussion by hamzawain7** [fabricjs/fabric.js#6754] [Ideas]
I want to render buttons slightly differently on Hover and Active state. I think that is not achievable with current api.

My proposition is to add mouseOverHandler, mouseOutHandler functions and isActive property

**Comment by asturur** [comment]
Why using fabricjs for things like buttons? does it make sense?

Anyway you already have mouseenter and mouseleave events ( i can't remember the exact names right now, but are in the docs and events demo )

**Reply by hamzawain7** [reply]
Sorry i meant i want to render the controls slightly differently on hover/active states. By button i meant a hovering delete button for an item like in your demo (http://fabricjs.com/custom-control-render). 

There isn't any mouseenter/mouseleave event in the control.class (would be nice to have those). I tried using the existing provided mousedown and mouseup for detecting whether a control was active but that can't be used as mouseup doesn't fire if mouse is outside the control

**Reply by asturur** [reply]
Sorry for not replying for a long time, in the hover event, there is always a reference to the control currently hovered, is the same reference we use to change the cursor style.
Object.__corner keep the value of the hovered control.
At that point you can change control properties to look different.
